src/PostgREST/AppState.hs

```diff
@@ -111,10 +113,17 @@ destroy = destroyPool
 
 initPool :: AppConfig -> IO SQL.Pool
 initPool AppConfig{..} =
-  SQL.acquire configDbPoolSize timeoutMicroseconds $ toUtf8 configDbUri
+  SQL.acquire configDbPoolSize timeoutMicroseconds configDbUriAndAppName
   where
     timeoutMicroseconds = (* oneSecond) <$> configDbPoolAcquisitionTimeout
     oneSecond = 1000000
+    configDbUriAndAppName = toUtf8 configDbUri <>
+      case uriQuery <$> parseURI (toS configDbUri) of
+        Nothing  -> " " <> appNameParam
+        Just ""  -> "?" <> appNameParam
+        Just "?" -> appNameParam
+        _        -> "&" <> appNameParam
+    appNameParam = "fallback_application_name='PostgREST " <> prettyVersion <> "'"
```

Comment:
I think a pretty print could be used instead of kebab-case. For instance, my setup returns this when doing:

```sql
select distinct application_name from pg_stat_activity;
```

|application_name|
|-|
|IntelliJ IDEA 2022.2.3|
|PostgREST 10.2.0.20230209 (pre-release) (16f2849)|

It looks good, only the commit hash and perhaps the "(pre-release)" should go.